### PR TITLE
Use exponential backoff for bot deploy retries

### PR DESCRIPTION
**Pull request by codyebberson**
Still occasionally running into these errors:

```
[Error]: The operation cannot be performed at this time. An update is in progress for resource: arn:aws:lambda:us-east-1:647991932601:function:medplum-bot-lambda-...
```

Before, we used constant 1 second retry interval.

After, using exponential backoff of 1 sec, 2 sec, 4 sec, etc.

### CODE REVIEW

File: packages/server/src/fhir/operations/deploy.ts [hunk 1]
```diff
@@ -32,7 +32,6 @@ export async function deployHandler(req: FhirRequest): Promise<FhirResponse> {
   }
 
   try {
-    const code = req.body.code as string | undefined;
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/deploy.ts:35]: Drive-by fix.  This is the exact same as line 25 above, so this line is redundant and confusing.

**Review by ThatOneBro — APPROVED** [review]
